tools/snp.sh

```diff
@@ -786,14 +839,19 @@ setup_and_launch_guest() {
   #add_qemu_cmdline_opts "-machine pc-q35-7.1"
 
   # snp object and kernel-hashes on
-  add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,kernel-hashes=on"
+  if [ "$SVSM" = true ]; then
+     add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,kernel-hashes=on"
+  else
+     add_qemu_cmdline_opts "-object memory-backend-memfd,size=2G,id=mem0,share=true,prealloc=false,reserve=off"
+     add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,init-flags=5,igvm-file="$IGVM_FILE""
+  fi
```

Comment:
Again, your logic seems reversed here?

Reply:
In what context is this applicable? Could you explain further? The else part is for SVSM, and we need to understand how to correctly configure the QEMU command for SVSM. There is slight differences in snp and svsm qemu commands.

Reply:
Your logic for the flag is backward. If SVSM is enabled, then do things for SVSM. If it is disabled, do things for the standard flow. Your logic is backward. That is what I mean.